### PR TITLE
Fix taxi2 docs and api-root for XSIAM and XSOAR SaaS

### PR DESCRIPTION
**Pull request by MosheEichler**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/XSUP-40987)

## Description
Fixed the api_root for XSIAM and XOSAR SaaS

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/TAXIIServer/Integrations/TAXII2Server/TAXII2Server.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def __init__(self, url_scheme: str, host: str, port: int, collections: dict, cer
             types_for_indicator_sdo: The list of stix types to provide indicator stix domain objects.
         """
         self._url_scheme = url_scheme
-        self._host = host
+        self._host = host.replace('.xdr.', '.crtx.')
```
Comment (MLainer1) [inline Packs/TAXIIServer/Integrations/TAXII2Server/TAXII2Server.py:66]: Is that relevant in all cases?

File: Packs/TAXIIServer/Integrations/TAXII2Server/TAXII2Server.py [hunk 2]
```diff
@@ -167,10 +167,16 @@ def get_discovery_service(self, instance_execute=False) -> dict:
             calling_context = get_calling_context()
             instance_name = calling_context.get('IntegrationInstance', '')
             endpoint = requote_uri(os.path.join('/instance', 'execute', instance_name))
-            service_address = f'{self._url_scheme}://{self._host}{endpoint}'
+            if is_xsiam_or_xsoar_saas():
```
Comment (MLainer1) [inline Packs/TAXIIServer/Integrations/TAXII2Server/TAXII2Server.py:170]: Are you sure we are not handling it in other places as well?

Comment (MosheEichler) [reply Packs/TAXIIServer/Integrations/TAXII2Server/TAXII2Server.py:170]: Not sure but not what I can found.
And without this changed the url is incorrect but with this change it's fine.

File: Packs/TAXIIServer/Integrations/TAXII2Server/TAXII2Server.py (hunk 1 above)
Comment (MosheEichler) [reply Packs/TAXIIServer/Integrations/TAXII2Server/TAXII2Server.py]: Per the [docs](https://github.com/demisto/content/blob/master/Packs/TAXIIServer/Integrations/TAXII2Server/TAXII2Server_description.md#how-to-access-the-taxii-service) as I understood yes.
LMK if you think different.
Screenshot 2024-09-08 at 23 53 53

**Review by thefrieddan1 — APPROVED** [review]

**Comment by MosheEichler** [comment]
Needs a force merge since it fails on internal server error which is a [known issue](https://jira-dc.paloaltonetworks.com/browse/CRTX-93966) and fixed by @saarmornel on XOSAR version 8.8 (currently the xsoar-ng build machines are on version 8.7)

**Pull request merged**